nix-buffer.el

```diff
@@ -216,9 +216,10 @@ is removed."
   (interactive)
   (let* ((root (directory-file-name (or (buffer-file-name) default-directory)))
 	 (expr-dir (locate-dominating-file root "dir-locals.nix")))
-    (when expr-dir
-	 (let ((expr-file (f-expand "dir-locals.nix" expr-dir)))
-	   (nix-buffer--nix-build root expr-file)))))
+    (if expr-dir
+	(let ((expr-file (f-expand "dir-locals.nix" expr-dir)))
+	  (nix-buffer--nix-build root expr-file))
+      (error "No dir-locals.nix file found"))))
```

Comment:
Ah or maybe I should just be using projectile here? I'm not familiar with it, will it let me call `nix-buffer` when and only when I open a file that has a dir-locals somewhere up the tree?